src/zjs_util.h

```diff
@@ -551,24 +551,26 @@ void zjs_loop_init(void);
 //     void remove(list_item_t *to_remove) {
 //         ZJS_LIST_REMOVE(list_item_t, my_list, to_remove);
 //     }
-#define ZJS_LIST_REMOVE(type, list, p)   \
-    ({                                   \
-        u8_t removed = 0;                \
-        type *cur = list;                \
-        if (p == list) {                 \
-            list = p->next;              \
-            removed = 1;                 \
-        } else {                         \
-            while (cur->next) {          \
-                if (cur->next == p) {    \
-                    cur->next = p->next; \
-                    removed = 1;         \
-                    break;               \
-                }                        \
-                cur = cur->next;         \
-            }                            \
-        }                                \
-        removed;                         \
+#define ZJS_LIST_REMOVE(type, list, p)       \
+    ({                                       \
+        u8_t removed = 0;                    \
+        if (p != NULL && list != NULL) {     \
```

Comment:
This should never be called with NULL. We could add an assert here to check these so that it would only run in debug mode, but we shouldn't do extra checks, we should document/enforce the calling specification.

Was this actually getting called with NULL? Why? That's the real bug.

Reply:
I wasn't actually having things call with NULL, just noticed there's no check and it would fail if passed a NULL list.

Reply:
@grgustaf Quickly grepping for all the code that calls it, none of it checks if the list is NULL before calling it, so we should make sure that its not possible that the list could be NULL at that point. Rather if we don't want a NULL check here, we should review the code that calls this macro to make sure its safe.